### PR TITLE
make user iden an in-band storm option and clean up API/tests

### PR DESCRIPTION
**Pull request by invisig0th**

### CODE REVIEW

File: synapse/lib/cell.py [hunk 1]
```diff
@@ -363,6 +359,14 @@ def setCellUser(self, iden):
         mesg = 'getRoleInfo denied for non-admin and non-member'
         raise s_exc.AuthDeny(mesg=mesg)
 
+    @adminapi
```
Comment (invisig0th) [inline synapse/lib/cell.py]: These were added to facilitate optic's need for checking perms and adding users to a local core or a core proxy

File: synapse/daemon.py [hunk 2]
```diff
@@ -325,6 +321,10 @@ def share(self, name, item):
         if finis:
             await asyncio.wait(finis)
 
+        for _, share in self.shared.items():
```
Comment (vEpiphyte) [inline synapse/daemon.py]: <3

File: synapse/cortex.py [hunk 3]
```diff
@@ -2594,82 +2617,103 @@ def getCoreMods(self):
...
-        async def nodes():
-            return [n async for n in self.eval(text, opts=opts, user=user)]
-        task = self.schedCoro(nodes())
-        return await task
-
-    @s_coro.genrhelp
-    async def streamstorm(self, text, opts=None, user=None):
-        '''
-        Evaluate a storm query and yield result messages.
+        if self.isfini: # pragma: no cover
```
Comment (jnwatson) [inline synapse/cortex.py]: There's no reason to make that no cover.

File: synapse/lib/agenda.py [hunk 4]
```diff
@@ -755,7 +755,8 @@ def list(self):
             logger.info('Agenda executing for iden=%s, user=%s, query={%s}', appt.iden, user.name, appt.query)
             starttime = time.time()
             try:
-                async for _ in self.core.eval(appt.query, user=user):  # NOQA
+                opts = {'user': user.iden}
+                async for _ in self.core.eval(appt.query, opts=opts):  # NOQA
```
Comment (jnwatson) [inline synapse/lib/agenda.py]: Why does this have #NOQA on it?

File: synapse/tests/test_lib_storm.py [hunk 5]
```diff
@@ -299,16 +298,16 @@ class StormTest(s_t_utils.SynTest):
...
-                                  core.eval('test:guid | max test:newp'))
-            await self.agenraises(s_exc.BadSyntax,
-                                  core.eval('test:guid | min test:newp'))
+            with self.raises(s_exc.BadSyntax):
+                await core.nodes('test:guid | max test:newp')
+            with self.raises(s_exc.BadSyntax):
+                await core.nodes('test:guid | min test:newp')
 
-            # Ensure that max evaluates ival properties as the upper bound.
+            # Ensure that max nodesuates ival properties as the upper bound.
```
Comment (jnwatson) [inline synapse/tests/test_lib_storm.py]: search/replace mistake

File: synapse/cortex.py [hunk 6]
```diff
@@ -403,48 +412,62 @@ def getCoreInfo(self):
         Yields:
             ((str,dict)): Storm messages.
         '''
-        view = self.cell._viewFromOpts(opts, self.user)
+        opts = self._reqValidStormOpts(opts)
 
         if opts is not None and opts.get('spawn'):
+            await self._execSpawnStorm(text, opts)
```
Comment (vEpiphyte) [inline synapse/cortex.py]: the precondition above for ``opts is not None`` is no longer true with `_reqValidStormOpts` and can be removed.

File: synapse/cortex.py [hunk 7]
```diff
@@ -2594,82 +2617,103 @@ def getCoreMods(self):
...
+            viewiden = user.profile.get('cortex:view')
+
+        if viewiden is None:
+            viewiden = self.view.iden
 
-        view = self.getView(iden=viewiden, user=user)
+        # For backwards compatibility, resolve references to old view iden == cortex.iden to the main view
+        # TODO:  due to our migration policy, remove in 0.3.x
+        if viewiden == self.iden: # pragma: no cover
+            viewiden = self.view.iden
```
Comment (vEpiphyte) [inline synapse/cortex.py]: `s_common.deprecated(....)` inside of this no-cover block.

File: synapse/cortex.py [hunk 8]
```diff
@@ -2594,82 +2617,103 @@ def getCoreMods(self):
...
+            return self.auth.rootuser
+
+        useriden = opts.get('user')
+        if useriden is None:
+            return self.auth.rootuser
+
+        user = self.auth.user(useriden)
+        if user is None:
+            mesg = f'No user found with iden: {opts.get("user")}'
+            raise s_exc.NoSuchUser(mesg)
```
Comment (vEpiphyte) [inline synapse/cortex.py]: iden=useriden ?

File: synapse/tests/test_cortex.py [hunk 9]
```diff
@@ -3764,9 +3792,10 @@ class Bawk:
                 for ($offs, $item) in $q.gets(size=1) { $q.cull($offs) }
             ''')
 
-            await asyncio.sleep(0)
+            # TODO figure out a way to fix this
+            await asyncio.sleep(0.1)
```
Comment (vEpiphyte) [inline synapse/tests/test_cortex.py]: ??????????????????????/

File: synapse/tests/test_lib_spawn.py [hunk 10]
```diff
@@ -305,12 +305,11 @@ class CoreSpawnTest(s_test.SynTest):
                 self.true(await asyncio.wait_for(evnt.wait(), timeout=6))
                 tasks = await prox.ps()
                 new_idens = [task.get('iden') for task in tasks]
-                self.len(1, new_idens)
-                await prox.kill(new_idens[0])
+                await prox.kill(new_idens[-1])
```
Comment (vEpiphyte) [inline synapse/tests/test_lib_spawn.py]: why remove the len check? ensured we have no stragglers

File: synapse/tests/test_lib_spawn.py [hunk 11]
```diff
@@ -305,12 +305,11 @@ class CoreSpawnTest(s_test.SynTest):
...
 
                 # Ensure that opts were passed into the task data without spawn: True set
                 task = [task for task in tasks if task.get('iden') == new_idens[0]][0]
-                self.eq(task.get('info').get('opts'), {'vars': {'hehe': 'haha'}})
+                self.none(task.get('info').get('opts').get('spawn'))
```
Comment (vEpiphyte) [inline synapse/tests/test_lib_spawn.py]: ? this change is ?

File: synapse/tests/test_lib_view.py [hunk 12]
```diff
@@ -179,5 +179,5 @@ class ViewTest(s_t_utils.SynTest):
             self.eq(1, count['node'])
             self.eq(1, count['fini'])
 
-            await self.agenraises(s_exc.BadConfValu,
-                                  core.streamstorm('[test:str=foo3 :hehe=bar]', opts={'editformat': 'jsonl'}))
+            with self.raises(s_exc.BadConfValu):
```
Comment (vEpiphyte) [inline synapse/tests/test_lib_view.py]: stormlist raises exceptions?

Comment (invisig0th) [reply synapse/tests/test_lib_view.py]: yep! :)

**Pull request merged**